dracut/30afterburn/afterburn-network-kargs.service

```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Afterburn Initrd Setup Network Kernel Arguments
+
+# This service may produce additional kargs fragments,
+# which are then consumed by dracut-cmdline(8).
+DefaultDependencies=no
+Before=dracut-cmdline.service
```

Comment:
is there anything complicated that needs to happen before this service runs (i.e. device loading or anything to query information (like VMware guestinfo)? If so this might be pretty early.

Reply:
For VMware specifically, no (it goes via an IOPort, the logic here does not require the vm-tool daemon running).

In general yes, this is very-painfully-early. But this needs to run before the cmdline is assembled by dracut and then fed into the initqueue networking.

I didn't find any better locations given the current status of things, do you have alternative places?

Reply:
This is fine for now. If we hit a constraint in the future we can do what I do in the copy_firstboot_network code where we [run after cmdline, but before initqueue](https://github.com/coreos/fedora-coreos-config/blob/a8c84a3e055500ce8119535240f17e9e78fb1f4e/overlay.d/05core/usr/lib/dracut/modules.d/15coreos-copy-firstboot-network/coreos-copy-firstboot-network.service#L7-L21) and then just [blank out the results of the earlier call to nm-initrd-generator](https://github.com/coreos/fedora-coreos-config/blob/a8c84a3e055500ce8119535240f17e9e78fb1f4e/overlay.d/05core/usr/lib/dracut/modules.d/15coreos-copy-firstboot-network/coreos-copy-firstboot-network.sh#L52-L54). If we switch to that model we'll need to add some more ordering to the copy_firstboot_network service to run after afterburn.

Reply:
(Or, rework NM to run as a unit and not in the initqueue)